src/filters.rs

```diff
@@ -497,6 +567,81 @@ pub fn strip_html(input: &Value, _args: &[Value]) -> FilterResult {
     }
 }
 
+/// Removes any newline characters (line breaks) from a string.
+pub fn strip_newlines(input: &Value, args: &[Value]) -> FilterResult {
+    try!(check_args_len(args, 0));
+    match *input {
+        Str(ref s) => Ok(Str(s.replace("\n", "").to_string())),
+        _ => Err(InvalidType("Str expected".to_string())),
+    }
+}
+
+pub fn times(input: &Value, args: &[Value]) -> FilterResult {
+    let num = match *input {
+        Num(n) => n,
+        _ => return Err(InvalidType("Num expected".to_owned())),
+    };
+    match args.first() {
+        Some(&Num(x)) => Ok(Num(num * x)),
+        _ => Err(InvalidArgument(0, "Num expected".to_owned())),
+    }
+}
+
+/// `truncate` shortens a string down to the number of characters passed as a parameter.
+///
+/// Note that this function operates on [grapheme
+/// clusters](http://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries) (or *user-perceived
+/// character*), rather than Unicode code points.  Each grapheme cluster may be composed of more
+/// than one Unicode code point, and does not necessarily correspond to rust's conception of a
+/// character.
+///
+/// If the number of characters specified is less than the length of the string, an ellipsis
+/// (`...`) is appended to the string and is included in the character count.
+///
+/// ## Custom ellipsis
+///
+/// `truncate` takes an optional second parameter that specifies the sequence of characters to be
+/// appended to the truncated string. By default this is an ellipsis (`...`), but you can specify a
+/// different sequence.
+///
+/// The length of the second parameter counts against the number of characters specified by the
+/// first parameter. For example, if you want to truncate a string to exactly 10 characters, and
+/// use a 3-character ellipsis, use 13 for the first parameter of `truncate`, since the ellipsis
+/// counts as 3 characters.
+///
+/// ## No ellipsis
+///
+/// You can truncate to the exact number of characters specified by the first parameter and show no
+/// trailing characters by passing a blank string as the second parameter.
+pub fn truncate(input: &Value, args: &[Value]) -> FilterResult {
+    if args.len() < 1 || args.len() > 2 {
+        return Err(InvalidArgumentCount(format!("expected one or two arguments, {} given",
+                                                args.len())));
+    }
+
+    let num_chars = match args.first() {
+        Some(&Num(x)) if x > 0f32 => x as usize,
+        _ => return Err(InvalidArgument(0, "Positive number expected".to_string())),
+    };
+
+    let ellipsis = "...".to_string();
+    let append = match args.get(1) {
+        Some(&Str(ref x)) => x,
+        _ => &ellipsis,
```

Comment:
Ah damn I just noticed you took that from `truncatewords`. Can you please fix this up in `truncatewords`, too? Thank you!